### PR TITLE
add ci for python 3.11 noray

### PR DESCRIPTION
**Pull request by stahanna**
**Description**

This pull request adds python 3.11 in the ci 

**Dear reviewer**

### CODE REVIEW

**Comment by registerrier** [comment]
Hi @stahanna , can you rebase now that #4969  has been merged?

**Review by registerrier — COMMENTED** [review]
Thanks @stahanna . This looks good.

The remaining issue is here https://github.com/gammapy/gammapy/actions/runs/7473558151/job/20337950158?pr=4998#step:6:4
setuptools is no longer part of stdlib in python 3.12 so that this command in ci.yml should be removed:
https://github.com/gammapy/gammapy/blob/4df020280a00c6944dbe890fe275b4c4f7ca81d4/.github/workflows/ci.yml#L85

**Comment by stahanna** [comment]
I removed python 3.12 but it looks as if there were sphinx problems now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,6 +39,15 @@ jobs:
...
             allowed_fail: false
+          - os: ubuntu-latest
+            python: '3.11'
+            tox_env: 'py311-test-alldeps_noray'
+            gammapy_data_path: /home/runner/work/gammapy/gammapy/gammapy-datasets/dev
+            allowed_fail: false
+#          - os: ubuntu-latest
+#            python: '3.12' # fail for the time being, to be tested later
+#            tox_env: 'py312-test'
+#            allowed_fail: false
```
Comment (registerrier) [inline .github/workflows/ci.yml:50]:
```suggestion
```

**Review by registerrier — APPROVED** [review]
Thanks @stahanna . The remaining fail is unrelated. It appears in other PRs as well.

I just removed the comments.

**Pull request merged**